src/appkit.rs

```diff
@@ -2305,3 +2306,54 @@ impl NSScreen for id {
         msg_send![self, convertRectToBacking:aRect]
     }
 }
+
+pub trait NSImage {
+    unsafe fn alloc(_: Self) -> id {
+        msg_send![class("NSImage"), alloc]
+    }
+
+    unsafe fn initByReferencingFile_(self, file_path: id /* (NSString *) */) -> id;
+}
+
+impl NSImage for id {
+    unsafe fn initByReferencingFile_(self, file_path: id /* (NSString *) */) -> id {
+        msg_send![self, initByReferencingFile:file_path]
+    }
+}
+
+pub trait NSButton {
+     unsafe fn setImage_(self, img: id /* (NSImage *) */);
+}
+
+impl NSButton for id {
+    unsafe fn setImage_(self, img: id /* (NSImage *) */) {
+        msg_send![self, setImage:img]
+    }
+}
+
+pub trait NSStatusItem {
+    unsafe fn button(self) -> id /* (NSStatusBarButton *) */;
+}
+
+impl NSStatusItem for id {
+    unsafe fn button(self) -> id /* (NSStatusBarButton *) */ {
+        msg_send![self, button]
+    }
+}
+
+pub const NSVariableStatusItemLength: CGFloat = -1.0;
+pub const NSSquareStatusItemLength: CGFloat = -2.0;
+
+pub trait NSStatusBar {
+    unsafe fn systemStatusBar(_: Self) -> id {
+        msg_send![class("NSStatusBar"), systemStatusBar]
+    }
+
+    unsafe fn statusItemWithLength_(self, length: CGFloat) -> id /* (NSStatusItem *) */;
```

Comment:
@jdm the convention seems to be replace `:` with `_` (or methods that take an argument) so this seems correct? the method is `statusItemWithLength:`

the apple reference is [here] (https://developer.apple.com/library/prerelease/mac/documentation/Cocoa/Reference/ApplicationKit/Classes/NSStatusBar_Class/index.html#//apple_ref/occ/instm/NSStatusBar/statusItemWithLength:)